node/external/transactionAPI/unmarshaller.go

```diff
@@ -110,6 +115,8 @@ func (tu *txUnmarshaller) prepareNormalTx(tx *transaction.Transaction) (*transac
 		GasLimit:         tx.GasLimit,
 		Data:             tx.Data,
 		Signature:        hex.EncodeToString(tx.Signature),
+		Options:          tx.Options,
+		Version:          tx.Version,
```

Comment:
the chainID should be filled also here from the internalTx if it exists.

Reply:
Right, only use for normal txs. Refactored

Reply:
ok